### DISCUSSION TITLE
Using Discovery Input helper through Pygeometry

### DISCUSSION THREAD

**Discussion by hdar1095** [ansys/pyansys-geometry#1565] [General]
When running scripts is run in Discovery  using :  result = modeler.run_discovery_script_file(
file_path="path/to/script.py",
script_args={},
)

If the script has discovery input helper, the script fails as it doesn't run it in debug mode. How do we run it in debug mode?

**Comment by RobPasMue** [comment]
Hi @hdar1095! Thanks for opening this discussion. To better assist you, could you share with us the following information?

1. What Ansys Discovery version are you using?
2. What PyAnsys Geometry version are you using (``pip list`` and search for ``ansys-geometry-core``)
3. Could you share a minimum Discovery script that you are trying to run using ``run_discovery_script_file()`` so that we can reproduce it on our side?

Thanks in advance!

**Comment by hdar1095** [comment]
Hi @RobPasMue .

1. I'm using Ansys Discovery 2024 R2, api version -242
2. ansys-geometry-core version: 0.6.6
3. I've saved the below script in .py and .dscript format. both doesn't work. 
In discovery the error message i get is : Unable to pause script, must run in debugging mode to pause

code : 
pSelection = DiscoveryInputHelper.PauseAndGetInput("select face", DiscoveryInputHelper.CreateSelection("select face",SelectionFilter.Face, True))
primarySelection= pSelection.PrimarySelection
secondarySelection = Selection.Empty()
result = NamedSelection.Create(primarySelection, secondarySelection, "Selection1")
result_2 = NamedSelection.Rename("Selection1", "s")

**Comment by RobPasMue** [comment]
@b-matteo could you look into this?

**Reply by b-matteo** [reply]
Hi @hdar1095, @RobPasMue 
Thanks for reaching out,
Debug mode on run_discovery_script_file will be accessible into the next ansys-geometry-core major version (0.8.XX) and accessible in remote starting with Ansys Discovery 25 R1 version.
That being said, @hdar1095, i've noticed that you're calling _DiscoveryInputHelper.PauseAndGetInput("select face", DiscoveryInputHelper.CreateSelection("select face",SelectionFilter.Face, True))_ into the script you want to execute. This will cause both the PyAnsys Geometry and Discovery scripts execution to stop while they'll be waiting for manual inputs into Discovery GUI. That might as well be the workflow you're looking for, but i just want to point that out to be sure, as it's quite unusual in a remote scripting context.

**Comment by jmabshof** [comment]
@RobPasMue I'm not sure it makes sense to support this. The Input Helper is an interactive tool that runs in the UI of Discovery. PyGeometry isn't really the place for such a feature, IMO

**Reply by RobPasMue** [reply]
Agreed with @jmabshof - after having a look at the implications it's best if we do not allow it IMO.

**Reply by jmabshof** [reply]
Will you close, then?

**Reply by RobPasMue** [reply]
Yep! On it

**Comment by hdar1095** [comment]
Hi,

I've created some scripts where user input is unavoidable due to non-patterned features that needs to be selected to cleanup further into the script. Is there a way around ?

Thanks

**Reply by RobPasMue** [reply]
Hi @hdar1095 -- could you specify which operations you are performing via user input?
